src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/SubaccountProcessor.kt

```diff
@@ -124,11 +124,14 @@ internal open class SubaccountProcessor(
         state.calculated[CalculationPeriod.current] = subaccountCalculated
 
         val fills = parser.asTypedList<IndexerCompositeFillObject>(content["fills"])
-        state = processFills(
-            subaccount = state,
-            payload = fills,
-            reset = false,
-        )
+        // fills is sometimes null on this message, doesn't mean there are none
+        if (fills != null) {
```

Comment:
I think this is why the fills were disappearing earlier too. This message has null fills even though there are some.

Reply:
reset = false is passed here, so it shouldn't overwrite the existing fills when payload is null.

Reply:
even when it's true fills is null though

Reply:
I mean processFills() handles the case when fills == null already... if fill == null, the existing fills won't be overwritten

Reply:
```
subaccount.fills = fillsProcessor.process(
            existing = if (reset) null else subaccount.fills,
            payload = payload ?: emptyList(),
            subaccountNumber = subaccount.subaccountNumber,
        )
```
        
 if reset is true it's going to nuke it, which we don't want because payload is also null